src/main/java/seedu/duke/Duke.java

```diff
@@ -0,0 +1,48 @@
+package seedu.duke;
+
+import java.io.IOException;
+
+import seedu.duke.Ui.Ui;
+import seedu.duke.utils.Parser;
+import seedu.duke.utils.Storage;
+import seedu.duke.utils.TaskList;
+
+/**
+ * main Duke class
+ */
+public class Duke {
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+    private Parser parser;
+
+    /**
+     * Duke constructor
+     *
+     * @param filePath filepath to duke.txt
+     */
+    public Duke(String filePath) {
+        this.storage = new Storage(filePath);
+        this.parser = new Parser(this.storage);
+        this.ui = new Ui();
+        try {
+            this.taskList = storage.readFile(this.ui);
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+    }
+
+
+    /**
```

Comment:
For javadoc comments you can consider writing comments for more functions and also giving @param and @return tags but I think should be fine since the rubrics only needs 50% to have javadocs